File: app/src/main/java/com/santalu/myapplication/MainActivity.java
package com.santalu.myapplication;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.ArrayAdapter;
import android.widget.Toast;
import com.santalu.respinner.ReSpinner;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    final ReSpinner spinner = findViewById(R.id.spinner);

    ArrayList<String> items = new ArrayList<>();
    for (int i = 0; i < 10; i++) {
      items.add(String.format("Item %s", i));
    }

    spinner.setAdapter(new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, items));
    spinner.setOnItemClickListener((adapterView, view, i, l) -> {
      String message = String.format("%s clicked", adapterView.getSelectedItem());
      Log.v("DEBUG", message);
      Toast.makeText(MainActivity.this, message, Toast.LENGTH_SHORT).show();
    });

    //this will call onItemClick
    //spinner.setSelection(3);

    //this will not call onItemClick
    //spinner.setSelection(3, false);
  }
}

File: app/src/main/java/com/santalu/sample/MainActivity.kt
package com.santalu.sample

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.ArrayAdapter
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.spinner

class MainActivity : AppCompatActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    val items = ArrayList<String>()
    for (i in 0..10) {
      items.add("Item $i")
    }

    spinner.adapter = ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item, items)
    spinner.setOnItemClickListener { parent, view, position, id ->
      Toast.makeText(
        this,
        "${parent.selectedItem} clicked",
        Toast.LENGTH_SHORT
      ).show()
    }

    // this will trigger onItemClick
    // spinner.setSelection(3)

    // this will not trigger onItemClick
    // spinner.setSelection(3, false)
  }
}
